app/helpers/enroller_query_helper.rb

```diff
@@ -0,0 +1,158 @@
+
+    # Filter by search text
+    patients = filter_by_text(patients, query[:search])
+
+    # Sort
+    sort(patients, query[:order], query[:direction])
+  end
+
+  def filter_by_text(patients, search)
+    return patients if search.nil? || search.blank?
+
+    filtered = patients.where('first_name like ?', "#{search&.downcase}%").or(
+      patients.where('last_name like ?', "#{search&.downcase}%").or(
+        patients.where('user_defined_id_statelocal like ?', "#{search&.downcase}%").or(
+          patients.where('user_defined_id_cdc like ?', "#{search&.downcase}%").or(
+            patients.where('user_defined_id_nndss like ?', "#{search&.downcase}%").or(
+              patients.where('date_of_birth like ?', "#{search&.downcase}%").or(
+                patients.where('patients.email like ?', "#{search&.downcase}%")
+              )
+            )
+          )
+        )
+      )
+    )
+
+    phone_query = search.delete('^0-9')
+    phone_query.blank? ? filtered : filtered.or(patients.where('primary_telephone like ?', "+1#{phone_query}%"))
+  end
+
+  def sort(patients, order, direction)
+    return patients if order.blank? || direction.blank?
+
+    # Satisfy brakeman with additional sanitation logic
+    dir = direction == 'asc' ? 'asc' : 'desc'
+
+    case order
+    when 'name'
+      patients = patients.order(last_name: dir, first_name: dir, id: dir)
+    when 'jurisdiction'
+      patients = patients.includes(:jurisdiction).order('jurisdictions.name ' + dir, id: dir)
+    when 'assigned_user'
+      patients = patients.order(Arel.sql('assigned_user IS NULL, assigned_user ' + dir), id: dir)
+    when 'state_local_id'
+      patients = patients.order(Arel.sql('user_defined_id_statelocal IS NULL OR user_defined_id_statelocal = "", user_defined_id_statelocal ' + dir), id: dir)
+    when 'sex'
+      patients = patients.order(Arel.sql('sex IS NULL OR sex = "", sex ' + dir), id: dir)
+    when 'dob'
+      patients = patients.order(Arel.sql('date_of_birth IS NULL, date_of_birth ' + dir), id: dir)
+    when 'enrollment_date'
+      patients = patients.order(Arel.sql('patients.created_at IS NULL, patients.created_at ' + dir), id: dir)
+    end
+
+    patients
+  end
+
+  def enroller_linelist(patients)
+    # get a list of fields relevant only to enroller table
+    fields = %i[name jurisdiction assigned_user state_local_id sex dob enrollment_date]
+
+    # retrieve proper jurisdiction
+    patients = patients.joins(:jurisdiction)
+
+    # only select patient fields necessary for enroller table
+    patients = patients.select('patients.id, patients.first_name, patients.last_name, patients.user_defined_id_statelocal, patients.sex, '\
+                               'patients.date_of_birth, patients.assigned_user, patients.created_at, jurisdictions.name AS jurisdiction_name, '\
+                               'jurisdictions.path AS jurisdiction_path, jurisdictions.id AS jurisdiction_id')
```

Comment:
`jurisdiction_path` and `jurisdiction_id` can be removed here and `jurisdiction_id` can be removed from the `linelist` method in `patient_query_helper`

Reply:
Removing `jurisdiction_id` breaks the sort, but I removed `jurisdiction_path`: https://github.com/SaraAlert/SaraAlert/pull/1211/commits/6f78a547f6ea4a4df6cb8dcdfbf51a388ab1de44